### PR TITLE
Refactor feeds api by using django rest framework serializers

### PR DESCRIPTION
**Pull request by UjjwalAggarwal-1**
# Description

Used serializers in django rest framework and refactored the feeds api to use this for serialization/deserialzation

Please include a summary of the change.
Created a FeedSerializer, updated the code in the feed api (inside api/views.py) accordingly

## Related issues
Please add related issues.
[#79](https://github.com/intelowlproject/GreedyBear/issues/79)
## Type of change
Refacotoring
- [x] Please delete options that are not relevant.

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://greedybear.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [ ] I have added documentation of the new features.
- [x] The tests gave 0 errors.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://greedybear.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](./Contribute.md)).
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

**Comment by UjjwalAggarwal-1** [comment]
@mlodic could you approve the workflows

File: api/serializers.py [hunk 1]
```diff
@@ -33,3 +33,54 @@ def validate(self, data):
...
+        formats = ["csv", "json", "txt"]
+
+        if feed_type not in feed_choices:
+            raise serializers.ValidationError("Feed type is not valid")
+        if attack_type not in attack_types:
+            raise serializers.ValidationError("Attack type is not valid")
+        if age not in age_choices:
+            raise serializers.ValidationError("Age is not valid")
+        if format_ not in formats:
+            raise serializers.ValidationError("Format is not valid")
```
Comment (mlodic) [inline api/serializers.py]: this is cool but don't you think we could simplify this further by leveraging ChoiceField? https://www.django-rest-framework.org/api-guide/fields/#choicefield

**Comment by UjjwalAggarwal-1** [comment]
@mlodic could you tell me what is wrong here. 
--> the 3rd test is failing on "install requirements"
I have not added any requirement, used preexisting ones only. And the logs do not show any error message except "failing"

**Review by mlodic — CHANGES_REQUESTED** [review]
the base of your work is cool, thanks for doing that.

One thing: as maintainers, we would need proof that this works.
To do that, we kindly ask you to add related tests that verify that the serializer works as intended

**Comment by mlodic** [comment]
> @mlodic could you tell me what is wrong here. --> the 3rd test is failing on "install requirements" I have not added any requirement, used preexisting ones only. And the logs do not show any error message except "failing"

this seems Github Actions errors (OOM) unrelated to this pr. I'll force the task again

File: api/serializers.py (hunk 1 above)
Comment (UjjwalAggarwal-1) [reply api/serializers.py]: oh thats a perfect fit I'll change it

**Comment by UjjwalAggarwal-1** [comment]
@mlodic I do not have any significant experience writing tests, what should I test for this code?

**Comment by mlodic** [comment]
first you would need to try to merge conflicts with the upstream. This may happen if another developer is adding new content on the `develop` branch.

then, you could take example from IntelOwl tests like [this](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_serializers.py) or [this](https://github.com/intelowlproject/IntelOwl/blob/develop/tests/api_app/test_serializers.py)

basically, you should run your new code with possible inputs and then verify that the serialization is performed correctly so for instance an input that should trigger ValidationError must trigger it...and so on

**Comment by UjjwalAggarwal-1** [comment]
@mlodic I have updated serializer with ChoiceField and also merged the latest upstream. I am working on test cases for serializer. Could you run the Github tests in the meanwhile.

**Comment by mlodic** [comment]
closed for inactivity
